### ISSUE TITLE
Luaj: Do not treat '_' as a SPACE character (Script error: =Module:ru-noun:1254 Too many arguments for argument set: arg 4 = c)

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#582]
I have come across this issue with `en.wiktionary.org/wiki/%D0%B1%D0%B0%D0%B1%D1%8C%D0%B5_%D0%BB%D0%B5%D1%82%D0%BE`
![underscore](https://user-images.githubusercontent.com/861631/65618533-5b480800-dfb6-11e9-98e4-327b4b09b29f.jpg)

working thru this I have found that Module:ru-noun is the cause
The actual area of issue is the local function to this module `ine`

    local function ine(arg)
      if not arg then return nil end
      arg = rsub(arg, "^%s*(.-)%s*$", "%1")
      if arg == "" then return nil end
      local inside_quotes = rmatch(arg, '^"(.*)"$')
      if inside_quotes then
        return inside_quotes
      end
      inside_quotes = rmatch(arg, "^'(.*)'$")
      if inside_quotes then
        return inside_quotes
      end
      return arg
    end

In this particular case `ine` is passed the following entries

    args { [1] = [[бабий|ба́бье]],[2] = +,[3] = _,[4] = c,[5] = [[ле́то]],["pos"] = noun,}  
and currently ends up with

    args { [1] = [[бабий|ба́бье]],[2] = +,[4] = c,[5] = [[ле́то]],["pos"] = noun,}  

Basically the character `_` is being lost
I suspect it is related to #571 (perhaps `_` is a special Lua escape string?)

**Comment by gnosygnu** [comment]
As always, thanks for the detailed breakdown. This was probably a tough issue to debug as ```ine()``` is a removed from the original error (```export.do_generate_forms```)

At any rate, this bug was related to #502 and is a bit significant. It treats ```_``` as a SPACE character (matching the Lua ```%s```). I thought this was part of the Java definition, but I checked now, and it's definitely not listed: https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/aa318070b27849f1fe00d14684b2a40f7b29bf79/jdk/src/share/classes/java/util/regex/UnicodeProp.java#L72-L75

Fixed with the commit above. Thanks for the detective work!
